### PR TITLE
Retry info / Advanced Configurations

### PR DESCRIPTION
**Pull request by dmarvp**
Adds a retry section as well as a special/advanced settings for e2e tests. 
I do think that we need to revisit the error codes before making this public. Seems to me that 550 and 551 could be merged as well as 553 and 554.

### CODE REVIEW

File: end-to-end/guide.md [hunk 1]
```diff
@@ -776,6 +768,32 @@ configuration:
...
+    "retryNumber" : 2
+}
+
+```
+
+`retryOn` is an array containing error codes returned from our Virtual Device server, while `retryNumber` is the number of times you want to retry your tests in case of an error (from 0 to 5, defaults at 2). Below is a list of the most significant error codes:
+
+Error code | Reason | Category
+-- | -- | --
+400 | A required parameter was set with invalid values. | user
```
Comment (jkelvie) [inline end-to-end/guide.md]: Remove the period at the end of the reason section, on this line and the others.

File: end-to-end/guide.md [hunk 2]
```diff
@@ -800,6 +818,20 @@ JEST_STARE_REPORT_TITLE
...
+
+| Key | Description |
+| --- | --- |
+| [asyncMode](#batch-or-sequential-tests) | Only works when batchEnabled is true, if asyncMode is false we wait for all the results, if it is true we retrieve them asynchronously - defaults to false |
+| [asyncE2EWaitInterval](#batch-or-sequential-tests) | Set an interval in milliseconds to wait before querying for new results, when batchEnabled is set to false - defaults to 5000 |
+| [batchEnabled](#batch-or-sequential-tests) | If it is true we sent the complete set of utterances to the virtual device server in a test, if it is false we sent them one by one - defaults to true |
+| ignoreExternalErrors | When a not controlled error happens while executing your tests, marks them as skipped instead of failures - defaults to `false` |
+| [maxAsyncE2EResponseWaitTime](#batch-or-sequential-tests) | Set an interval in milliseconds to wait before stop looking for new results, when batchEnabled is set to false - defaults to 15000 |
+| [retryOn](#retrying-tests) | An array with Virtual Device error codes on which to do a retry if a test fails |
```
Comment (jkelvie) [inline end-to-end/guide.md:831]: What is the default behavior regarding retries? Do we attempt retries by default?

**Review by jkelvie — CHANGES_REQUESTED** [review]

File: end-to-end/guide.md (hunk 2 above)
Comment (dmarvp) [reply end-to-end/guide.md:831]: We do attempt retries by default (codes 551 and 552). It's on line 796 but I can also add it here if you want.

Comment (jkelvie) [reply end-to-end/guide.md:831]: I missed line 796. I think that is sufficient

**Review by jkelvie — APPROVED** [review]

**Review by jperata — APPROVED** [review]

**Pull request merged**